File: space-game/OrbitalMechanics/CelestialObjects/IOrbitingObject.vb
Imports OrbitalMechanics.Classes

Namespace CelestialObjects
    Public Interface IOrbitingObject
        ReadOnly Property Orbit As IOrbit
        ReadOnly Property Primary As Guid
        Sub SetOrbit(p As BaseCelestialObject, o As IOrbit)
    End Interface
End NameSpace

File: space-game/OrbitalMechanics/Interfaces/IOrbitingObject.cs
using OrbitalMechanics.CelestialObjects;
using System;

namespace OrbitalMechanics.Interfaces
{
    public interface IOrbitingObject
    {
        IOrbit Orbit { get; }
        Guid Primary { get; }
        void SetOrbit(BaseCelestialObject p, IOrbit o);
    }
}
